salt/cert/init.sls

```diff
@@ -2,7 +2,8 @@ include:
   - crypto
 
 {% set ip_addresses = [] -%}
-{% set extra_names = ["DNS: " + grains['caasp_fqdn'] ] -%}
+{% set caasp_fqdn = grains['id'] + "." + pillar['internal_infra_domain'] -%}
+{% set extra_names = ["DNS: " + caasp_fqdn, "DNS: " + grains['nodename'] ] -%}
```

Comment:
As we have no requirement that hostnames are unique, each machine will have a cert that is valid for all machines, leaving us with a security risk.

Reply:
There is no security risk, we can enable this only for CPI.